private/signer/v4/v4_test.go

```diff
@@ -54,8 +54,8 @@ func TestPresignRequest(t *testing.T) {
 	signer.sign()
 
 	expectedDate := "19700101T000000Z"
-	expectedHeaders := "host;x-amz-meta-other-header;x-amz-target"
-	expectedSig := "5eeedebf6f995145ce56daa02902d10485246d3defb34f97b973c1f40ab82d36"
+	expectedHeaders := "content-type;host"
```

Comment:
I'd add tests for the x-amz-* in query string

and verify x-amz-meta-* are good in query string